projects/igniteui-angular/src/lib/grids/grid/grid.component.ts

```diff
@@ -145,6 +145,7 @@ export class IgxGridComponent extends IgxGridBaseDirective implements GridType,
     }
 
     public set data(value: any[]) {
+        this.crudService.endEditMode();
```

Comment:
I tested the PR and everything seem to be working fine but I just wonder is there any reason to use this method from the crud service instead of endEdit method in grid-base?

Reply:
calling `endEdit` will fire `cellEditExit` and `rowEditExit`. IMO we should not fire any events when exiting edit mode because of data source change. This is why I used more "direct" approach via `crudService`.